### DISCUSSION TITLE
[BUG] `'No module named  pyarmor_runtime_00000` when using obfuscated .py package coverted to wheel (.whl)

### DISCUSSION THREAD

**Discussion by Dtano17** [dashingsoft/pyarmor#1412] [General]
Please read this first before report any issue
https://pyarmor.readthedocs.io/en/latest/questions.html

Please copy the whole command line `pyarmor gen` and first 4 lines in the console, do not mask anything, and do not paste snapshot image

```
$ pyarmor gen -O dist test.py
INFO     Python 3.9.13
INFO     Pyarmor 8.2.9 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     search inputs ...   

```
I am having issues when using pyarmor 8 edition in python 3.9.13 runtime  when i try to distribute and reuse the obfuscated python package in a wheel format after following the steps outlined here:
https://pyarmor.readthedocs.io/en/stable/how-to/wheel.html?highlight=wheel 

after building and generating the .whl file successfully, I am not able to reuse the python package and a module in a different environment (for my use case, databricks python runtime environment) as it gives an error that it cannot find the pyarmor_runtime_00000 module regardless of if i uses the  -i flag or not during obfuscation and where I placed the pyarmor_runtime.pyd file in the  pyarmor_runtime_00000  folder.

Error: 'No module named  pyarmor_runtime_00000'

(see the screenshot for our package structure and some steps I followed).

However, when I use the pyarmor-7, it works seamlessly with databricks python runtime and works in all environments, it creates the _pytransform.dll file in the pytransform folder which works perfectly for my use case.

Question:
- How can I resolve the pyarmor 8 error using the runtime configuration to work on my current Databricks python runtime environment?

**Comment by jondy** [comment]
The wheel is build by yourself, please refer to wheel documentation to include package `pyarmor_runtime_000000` to wheel

**Comment by jondy** [comment]
Just take the obfuscated scripts as normal scripts to build wheel, for any import issues, check the obfuscated scripts `cat dist/xx.py` to find why it can't be imported.

**Reply by Dtano17** [reply]
@jondy just to further elaborate my point, after following the steps [here](https://pyarmor.readthedocs.io/en/stable/how-to/wheel.html?highlight=wheel):
I can successfully generate and build my wheel package and it works in the vs code command line interface to confirm that i am correctly following the steps as shown below:
![image](https://github.com/dashingsoft/pyarmor/assets/29021180/f97330a6-f626-443f-89ff-830741227746)

however, when i download this wheel file and install in databricks python runtime and try to use, it gives the error:
ModuleNotFoundError: No module named 'parent.pyarmor_runtime_000000.pyarmor_runtime' as shown below:
![image](https://github.com/dashingsoft/pyarmor/assets/29021180/086392d4-9dfc-4ce0-a300-0a8337596cd4)

side note, when this same packae when generated using the previous pyarmor-7 and distributed with wheel, it works seamlessly without any additional steps. 
So my questions are, is this a problem with the pyarmor-8 runtime change from _pytransform.dll to pyarmor_runtime.pyd, if so, when will this be resolved, can i use pyarmor-7(for the main time) or how can i can resolve this issue if not.

Thanks in advance for your swift response.

**Reply by MegalaRamu** [reply]
@Dtano17 Am also facing the same issue. Was anyone able to resolve this issue?
